pandas/core/internals/blocks.py

```diff
@@ -1774,8 +1774,8 @@ def to_native_types(self, slicer=None, na_rep="nan", quoting=None, **kwargs):
         mask = isna(values)
 
         try:
-            values = values.astype(str)
             values[mask] = na_rep
+            values = values.astype(str)
```

Comment:
i think u can move the astype outside of the tr/except

Reply:
👍🏼